uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/LeaveRequestSelect.php

```diff
@@ -102,6 +102,31 @@ private function addWhere(CRM_Utils_SQL_Select $customQuery) {
     if($hasUnassignedAsTrue) {
       $conditions[] = "NOT (" . implode(' AND ', $this->hasActiveLeaveManagerCondition()) . ") 
                        OR (r.is_active IS NULL AND rt.is_active IS NULL)";
+
+      $contactID = false;
+      $contactPassedAsArray = !empty($this->params['contact_id']['IN']);
+      $contactPassedAsID = isset($this->params['contact_id']) && is_numeric($this->params['contact_id']);
+
+      if($contactPassedAsArray) {
+        $contactID = $this->params['contact_id']['IN'];
+      }
+
+      if($contactPassedAsID) {
+        $contactID = [$this->params['contact_id']];
+      }
```

Comment:
Please extract this logic to get the contact ID from the params to a separate method